### PR TITLE
make link downloadable when it has downlod icon in it

### LINKED ISSUE #320: Autoblocking of downloadable images
make link as downloadable link when the link authored with download icon in it.

Ex: 

authoring :     
image

preview : 
image

### PR DESCRIPTION
**Pull request by rrrragas**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fixes #320 

Test URLs:
- Original: [https://www.sunstar.com/<path>](https://www.sunstar.com/newsroom/download/)
- Before: [https://main--sunstar--hlxsites.hlx.live/<path>](https://main--sunstar--hlxsites.hlx.page/_drafts/sagar/download)
- After: [https://$branch--sunstar--hlxsites.hlx.live/<path>](https://320-autoblocking-of-downloadable-images--sunstar--hlxsites.hlx.live/_drafts/sagar/download)

- [ ] New Blocks introduced in this PR
      If yes, please provide details below

Block Name    | Documentation
------------- | -------------
 For e.g. cards | [doc-link](https://main--sunstar--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/cards&index=0)

- [ ] New variations introduced in this PR

Variation Name    | Documentation
------------- | -------------
 For e.g. cards (grid)  | [doc-link](https://main--sunstar--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/cards&index=2)

### CODE REVIEW

**Comment by jindaliiita** [comment]
@rrrragas Can you please check link download variant. I think it serves the same purpose.

**Comment by rrrragas** [comment]
> @rrrragas Can you please check link download variant. I think it serves the same purpose.

> @jindaliiita yes , I just checked it. link with download variant does the same. I think we still need this for the links authored in other blocks (ex: download link in cards block)

File: scripts/scripts.js [hunk 1]
```diff
@@ -179,6 +179,17 @@ export function buildImageWithCaptionBlocks(main, buildBlockFunction) {
   });
 }
 
+function buildDownloadableLinks(main) {
```
Comment (sdmcraft) [inline scripts/scripts.js:182]: I think this should be a part of the anchor decoration flow i.e. `decorateAnchors/decorateExternalAnchors`. As a part of anchor decoration, you can invoke this method to decorate the downloadable links specially.

Comment (rrrragas) [reply scripts/scripts.js]: Done. Can you please check now

**Comment by dnbute** [comment]
>link with download variant does the same. I think we still need this for the links authored in other blocks (ex: download link in cards block)

Could we maybe remove that variant then and just have this autoblocking?

**Comment by rrrragas** [comment]
> > link with download variant does the same. I think we still need this for the links authored in other blocks (ex: download link in cards block)
> 
> Could we maybe remove that variant then and just have this autoblocking?

> Yes, as we are making any link that has download icon as downloadable link.  @jindaliiita can you please confirm

**Review by dnbute — APPROVED** [review]
LGTM, I'm not sure if anybody uses the download variant of the 'link' block, but if not, let's delete it

**Comment by jindaliiita** [comment]
> LGTM, I'm not sure if anybody uses any download variant, but if not let's delete it

Don't delete it as it is being used at various japanese news pages

**Comment by dnbute** [comment]
>Don't delete it as it is being used at various japanese news pages

Could they be replaced with a link? Could you share one of these pages?

**Comment by sdmcraft** [comment]
> Could they be replaced with a link? Could you share one of these pages?

The [finder tool](https://github.com/sdmcraft/web-tools/blob/main/docs/MARKUP-FINDER.md) tells me that `.link.download` selector is present across 343 pages.

image

Probably this can't be fixed manually and needs re-import. I'd suggest leaving the `.link.download` block variant as it is but mark it as `deprecated` in block library. Also log a ticket to remove it in future (and probably tag it as "Good to have".

**Comment by jindaliiita** [comment]
@rrrragas Can you please this in sidekick so that i can use it on new pages.

**Comment by jindaliiita** [comment]
Space seems to be less between text and download icon.
Screenshot 2023-11-13 at 9 28 14 PM

**Pull request merged**